src/main/java/edu/wpi/grip/core/Step.java

```diff
@@ -67,7 +67,7 @@ public void onInputSocketChanged(SocketChangedEvent e) {
         final Socket socket = e.getSocket();
 
         // If this socket that changed is one of the inputs to this step, run the operation with the new value.
-        if (socket.getStep() == this && socket.getDirection().equals(Socket.Direction.INPUT)) {
+        if (socket.getStep().isPresent() && socket.getStep().get() == this && socket.getDirection().equals(Socket.Direction.INPUT)) {
```

Comment:
You can do `socket.getStep().ifPresent(step ->`...  to avoid the manual check.

Another way of doing it is `if (socket.getStep().equals(Optional.of(this))`